index.js

```diff
@@ -63,16 +64,21 @@ module.exports  = function (params, callback) {
     var date = file.data.updated || file.data.date || new Date();
     var changefreq = file.data.changefreq || options.changefreq;
     var priority = file.data.priority || options.priority;
-
+    var filepath = getExternalFilePath(relativedest, file);
+    
     if (exclusion.indexOf(file.basename) !== -1 ||
         grunt.file.isMatch({srcBase: options.dest}, exclusion, file.dest)) {
       robots.push('Disallow: /' + getExternalFilePath(relativedest, file));
       return;
     }
 
+    if(options.removeindex === true) {
```

Comment:
I think you should just move this logic into `getExternalFilePath` function so it can be used like it was before.